### ISSUE TITLE
imf_ids() request fails

### ISSUE THREAD

**Issue by bdengler** [christophergandrud/imfr#11]
I am trying to retrieve `imf_ids()`, but get this response: 

```
Downloading: 860 B     Request failed [404]. Retrying in 1 seconds...
Downloading: 860 B     Request failed [404]. Retrying in 1.3 seconds...
Downloading: 860 B     Error: data.imf.org appears to be down.
```

I don't believe my machine/connection is the problem, as other commands, such as `imf_codes("CL_INDICATOR_IFS")`, do work. 

My `sessionInfo()`:

``` 
sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] imfr_0.1.5       bindrcpp_0.2     stringi_1.1.6    treemapify_2.4.0 treemap_2.4-2   
 [6] assertr_2.0.2.2  haven_1.1.1      forcats_0.2.0    stringr_1.2.0    dplyr_0.7.4     
[11] purrr_0.2.4      readr_1.1.1      tidyr_0.7.2      tibble_1.4.2     ggplot2_2.2.1   
[16] tidyverse_1.2.1 

loaded via a namespace (and not attached):
 [1] ggrepel_0.7.0      Rcpp_0.12.15       lubridate_1.7.1    lattice_0.20-35   
 [5] assertthat_0.2.0   digest_0.6.12      psych_1.7.8        utf8_1.1.3        
 [9] gridBase_0.4-7     mime_0.5           R6_2.2.2           cellranger_1.1.0  
[13] plyr_1.8.4         httr_1.3.1         pillar_1.1.0       rlang_0.1.6       
[17] curl_3.1           lazyeval_0.2.1     readxl_1.0.0       rstudioapi_0.7    
[21] data.table_1.10.4  labeling_0.3       foreign_0.8-69     htmlwidgets_1.0   
[25] igraph_1.1.2       munsell_0.4.3      shiny_1.0.5        broom_0.4.3       
[29] compiler_3.4.3     httpuv_1.3.5       modelr_0.1.1       pkgconfig_2.0.1   
[33] mnormt_1.5-5       htmltools_0.3.6    ggfittext_0.5.0    crayon_1.3.4      
[37] grid_3.4.3         nlme_3.1-131       jsonlite_1.5       xtable_1.8-2      
[41] gtable_0.2.0       magrittr_1.5       scales_0.5.0       cli_1.0.0         
[45] reshape2_1.4.2     xml2_1.2.0         RColorBrewer_1.1-2 tools_3.4.3       
[49] glue_1.2.0         hms_0.3            rsconnect_0.8.5    parallel_3.4.3    
[53] yaml_2.1.16        colorspace_1.3-2   rvest_0.3.2        knitr_1.19        
[57] bindr_0.1         
```

**Comment by cjyetman** [comment]
looks like the old endpoint here has gone dead: https://github.com/christophergandrud/imfr/blob/2bd7d0f96d94f35d73aae120bbaa29ecbadb4e22/R/data_functions.R#L20

possible new one here https://sdmxcentral.imf.org/ws/public/sdmxapi/rest/Dataflow?format=sdmx-json
but it gives database ids that are not available in the old endpoint, so this would require changes throughout the package

**Comment by cjyetman** [comment]
parsing would need to be different also, for example...
```r
URL <- "https://sdmxcentral.imf.org/ws/public/sdmxapi/rest/Dataflow?format=sdmx-json"
raw_dl <- imfr:::download_parse(URL)
data_id <- raw_dl$Dataflow$id
long_name <- unlist(sapply(raw_dl$Dataflow$names, `[`, "value"))
id_name <- data.frame(database_id = data_id, description = long_name)
```

**Comment by christophergandrud** [comment]
Solved by: 1086a23e1714da4c24a7851dfd77b0bcca7dfad4

Thanks @cjyetman !
